release/util.bzl

```diff
@@ -79,3 +82,88 @@ def sdk_tarball(name, version):
         """.format(version = version),
         visibility = ["//visibility:public"],
     )
+
+def _protos_zip_impl(ctx):
+    posix = ctx.toolchains["@rules_sh//sh/posix:toolchain_type"]
+    tmp_dir = ctx.actions.declare_directory("tmp_dir")
+    zipper_args_file = ctx.actions.declare_file(
+        ctx.label.name + ".zipper_args",
+    )
+    tools = [ctx.executable._tar, ctx.executable._gzip]
+    ctx.actions.run_shell(
+        inputs = [ctx.file._ledger_api_tarball] + ctx.files._daml_lf_tarballs,
+        outputs = [tmp_dir],
+        tools = tools,
+        command = """
+          set -eou pipefail
+          export PATH=$PATH:{path}
+          tar xf {ledger_api_tarball} -C {tmp_dir}
+          for file in {lf_tarballs}
+          do
+              tar xf $file -C {tmp_dir}
+          done
+        """.format(
+            ledger_api_tarball = ctx.file._ledger_api_tarball.path,
+            tmp_dir = tmp_dir.path,
+            lf_tarballs = " ".join([f.path for f in ctx.files._daml_lf_tarballs]),
+            path = ":".join(["$PWD/`dirname {tool}`".format(tool = tool.path) for tool in tools]),
+        ),
+    )
+
+    # zipper does not have an option to recursively zip files so we
+    # use find to list the files.
+    ctx.actions.run_shell(
+        outputs = [zipper_args_file],
+        inputs = [tmp_dir],
+        command = """
+        {find} -L {tmp_dir} -type f | {sed} -E 's#^{tmp_dir}/(.*)$#protos-{version}/\\1={tmp_dir}/\\1#' > {args_file}
```

Comment:
You could also use something like
```
{find} -L daml-lf/archive/src/main/protobuf -type f -printf "protos-{version}/%P=%p\n" > {args_file}
```
instead of using `sed`. Note the upper and lower-case `p`.

Reply:
Much nicer, thank you! Your `find` skillz are clearly ahead of mine :slightly_smiling_face: